### ISSUE TITLE
Line breaking happening at 96 print width?

### ISSUE THREAD

**Issue by vincerubinetti** [hosseinmd/prettier-plugin-jsdoc#168]
Not sure if this is related to #8 ?

95 characters:

<img width="1130" alt="image" src="https://user-images.githubusercontent.com/8326331/174350959-305682db-59a5-4a92-871b-5e24b67f47f7.png">

96 characters:

<img width="1120" alt="image" src="https://user-images.githubusercontent.com/8326331/174351088-3da20d62-e90d-4bd1-a0fc-46f8361869a6.png">

I'd expect it to wrap such that no characters (including the comment characters) exceed the print width (shown in the screenshot with the vertical line).

This is my `.prettierrc`:

```json
{
  "plugins": ["./node_modules/prettier-plugin-jsdoc"],
  "jsdocCapitalizeDescription": false,
}
```

No override of the default `printWidth` of `80`, so it should be `80`. Also the rest of my code formats properly to `80`.

**Comment by ahmafi** [comment]
I have a problem related to `printWidth` since it might be related to the same problem, I'll post it here for now. I get line break on 90-91 characters, but I have the default setting which is 80 characters. I have no other special options. I don't have this problem on the same line of the tag definition (line 8 and 10 in the image below).

![image](https://user-images.githubusercontent.com/12560109/183426120-68c7f382-0199-4240-b6d0-663eb047d426.png)

**Comment by vincerubinetti** [comment]
Does anyone know who the main maintainer(s) of this are? @hosseinmd? Any info about this? I don't know about others, but this line-wrapping is like the main point of using this extension at all, so it's not good that it doesn't work as expected.

I'd be happy to look into doing a PR for this. https://github.com/hosseinmd/prettier-plugin-jsdoc/commit/770e213f3cbf30f4dba0449fb2953c08a05a5bd7 seems to be a likely culprit, though I don't understand how it would create breaking at 96.

**Comment by vincerubinetti** [comment]
Something I just realized is that this bug seems to only happen if it's the last line of a comment:

<img width="966" alt="image" src="https://user-images.githubusercontent.com/8326331/183454004-bceeb8a5-f207-4113-accd-966600823c0e.png">

**Comment by vincerubinetti** [comment]
Okay, according to this comment, I guess this was actually a feature:

https://github.com/hosseinmd/prettier-plugin-jsdoc/issues/26#issuecomment-771823206

I was going to say that this was a weird error in judgement. I don't know of any word processor that tries to "smart" wrap like this. MS Word doesn't do it. VSCode's word wrap doesn't do it. Etc.

However, apparently Prettier does sort of do it:

> In other words, don’t try to use printWidth as if it was ESLint’s [max-len](https://eslint.org/docs/rules/max-len) – they’re not the same. max-len just says what the maximum allowed line length is, but not what the generally preferred length is – which is what printWidth specifies.

Despite all this, I wish there was a way to disable this behavior. I would like to just have it hard wrap at `printWidth` always.

Also, I can't verify this, but it seems like Prettier does actually hard wrap at `printWidth` whenever possible, but sometimes it's not able to, e.g a long string literal, hence why they say it sometimes produces lines over the `printWidth`.

**Comment by ahmafi** [comment]
I don't like this feature too, I prefer a new line even if it's only 1 word on the new line, so I have more consistency in my line widths. But since this is implemented and some people want this, maybe we can turn this into an option that can be configured.

**Comment by hosseinmd** [comment]
We need a paragraph balancer algorithm. Something which can keep lines smaller than print width, But about the same size

For Example:
```
Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a 
typeface withoutrelying on meaningful content.
```
Should be:
```
Lorem ipsum is a placeholder text commonly used to demonstrate the visual 
form of a document or a typeface without relying on meaningful content.
```

**Comment by vincerubinetti** [comment]
> We need a paragraph balancer algorithm.

Why? Does Prettier do this with prose? It certainly doesn't seem to do it with code. I think this is over-complicating it. I, and I think a lot of others, will just want dead simple line wrapping. Go line by line and start a new line the very second a line goes over the print width. Is it possible to get an option for this?

**Comment by hosseinmd** [comment]
Option for what?

**Comment by hosseinmd** [comment]
There is a package which @egill suggested https://github.com/nginz/fair-lines .

**Comment by vincerubinetti** [comment]
I'm saying, please just make it like this, or give us an option for this 👇 

> ```
> Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a |
> typeface withoutrelying on meaningful content.                                                    |
> ```

This is unnecessary 👇 

> ```
> Lorem ipsum is a placeholder text commonly used to demonstrate the visual                  |
> form of a document or a typeface without relying on meaningful content.                    |
> ```

**Comment by hosseinmd** [comment]
what about this
```
Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a 
all.
```

**Comment by egilll** [comment]
I have a fix for this issue (along with several other line-wrapping bugs) in a fork I've been working on, I'll try to submit it within 2 weeks.

**Comment by vincerubinetti** [comment]
Yes, that is fine, to me.

I understand the issue completely. I just want simple word wrapping, like most word processors do:

![Screen Recording 2022-08-11 at 11 17 10 AM (2)](https://user-images.githubusercontent.com/8326331/184169773-8dbea3d9-9926-4f2c-87a3-599625bd4018.gif)

I am willing to bet most other people are just looking for this too. I dont see a reason to make it so complicated with the "fair lines". It's cool if you want to add that, but an option for simple word wrapping would be greatly appreciated.

**Comment by hosseinmd** [comment]
> I have a fix for this issue (along with several other line-wrapping bugs) in a fork I've been working on, I'll try to submit it within 2 weeks.

Could you create a Quick PR just for this.

**Comment by hosseinmd** [comment]
> > what about this
> > ```
> > Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a 
> > all.
> > ```
> 
> Yes, that is fine, to me.
> 
> I understand the issue completely. I just want simple word wrapping, like most word processors do:
> 
> ![Screen Recording 2022-08-11 at 11 17 10 AM (2)](https://user-images.githubusercontent.com/8326331/184169773-8dbea3d9-9926-4f2c-87a3-599625bd4018.gif) [ ![Screen Recording 2022-08-11 at 11 17 10 AM (2)](https://user-images.githubusercontent.com/8326331/184169773-8dbea3d9-9926-4f2c-87a3-599625bd4018.gif) ](https://user-images.githubusercontent.com/8326331/184169773-8dbea3d9-9926-4f2c-87a3-599625bd4018.gif) [ ](https://user-images.githubusercontent.com/8326331/184169773-8dbea3d9-9926-4f2c-87a3-599625bd4018.gif)
> 
> I am willing to bet most other people are just looking for this too. I dont see a reason to make it so complicated with the "fair lines". It's cool if you want to add that, but an option for simple word wrapping would be greatly appreciated.

Ok. I Will do it as you want this weekend.

**Comment by vincerubinetti** [comment]
I'm sorry for being forceful. I'm guess I'm used to library developers making a baffling decision (in my opinion) and refusing to even consider changing it. But you are not like that. Thank you.

I am working on a PR now for this change. I'll also tag @egilll in it so that we can coordinate our fixes. I'm not familiar with this code base, so it might be a bad PR, but I should at least create a starting point, since I am the one requesting this change.

**Comment by hosseinmd** [comment]
Released v0.4.1
